magenta-lib/src/main/scala/magenta/tasks/AWS.scala

```diff
@@ -33,15 +33,14 @@ import software.amazon.awssdk.services.sts.model.AssumeRoleRequest.Builder
 import scala.annotation.tailrec
 import scala.collection.JavaConverters._
 import scala.util.{Random, Try}
-import magenta.withResource
 import software.amazon.awssdk.services.sts.auth.StsAssumeRoleCredentialsProvider
 import software.amazon.awssdk.services.sts.model.AssumeRoleRequest
 
 object S3 {
-  def withS3client[T](keyRing: KeyRing, region: Region, config: ClientOverrideConfiguration = AWS.clientConfiguration)(block: S3Client => T): T =
+  def withS3client[T](keyRing: KeyRing, region: Region, config: ClientOverrideConfiguration = AWS.clientConfiguration, resources: DeploymentResources)(block: S3Client => T): T =
```

Comment:
I'd consider putting the resources as the first argument here.

Reply:
I wish I saw this last week...have repeated the same pattern in every client builder 😭

Reply:
The difficulty with leaving it at the end is losing the behaviour of default parameters at the end of a parameter list. I see you've dealt with that by naming the param and this is only an issue for this one example due to the way the testing is done.